pkg/resource/deploy/deployment_executor.go

```diff
@@ -252,15 +279,34 @@ func (ex *deploymentExecutor) Execute(callerCtx context.Context, opts Options, p
 					return false, nil
 				}
 
-				if err := ex.handleSingleEvent(event.Event); err != nil {
-					if !result.IsBail(err) {
-						logging.V(4).Infof("deploymentExecutor.Execute(...): error handling event: %v", err)
-						ex.reportError(ex.deployment.generateEventURN(event.Event), err)
+				singleEventWorker := func() error {
+					if err := ex.handleSingleEvent(event.Event); err != nil {
```

Comment:
Since we're parallelizing at this level, we probably need some locking around the `skipped` mapset in the deployment executor as well.  

Which also makes me wonder, this does a little more than parallelizing the step generator, it parallelizes deployment event handling, so we might want to rename the PR?  This tripped me up a bit while reading through this PR the first time.

Reply:
Yeah, it's an odd shorthand. In order to run the step generator concurrently we need to add concurrency at the callsite which is in the deployment executor, I'll update the message

Reply:
> Since we're parallelizing at this level, we probably need some locking around the skipped mapset in the deployment executor as well.

Yes, I'll add some synchronization